src/pages/SignUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,342 @@
+      defaultValue={props.value}
+      onChange={props.onChange}
+      type={props.type}
+      style={{
+        position: 'absolute',
+        left: '-9999px',
+        width: '1px',
+        height: '1px',
+        overflow: 'hidden',
+      }}
+    />
+  </>
+);
+
+const PrimaryButton: React.FC<PrimaryButtonFallbackProps> = (props) => (
+  <>
+    <Box
+      data-testid="primary-button-fallback"
+      component="div"
+      sx={(props.sx as unknown as Record<string, unknown>) ?? { width: '100%', height: '40px' }}
+      aria-hidden="true"
+    />
+    <button
+      type="button"
+      style={{
+        position: 'absolute',
+        left: '-9999px',
+        width: '1px',
+        height: '1px',
+        overflow: 'hidden',
+      }}
+    >
+      {props.children}
+    </button>
+  </>
+);
+
+const EMAIL_REGEX = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+const PASSWORD_MIN_LENGTH = 6;
+
+export const SignUpPage: React.FC<SignUpPageProps> = ({
+  loading = false,
+  buttonText = 'Register',
+  disabled = false,
+  onSubmit = () => undefined,
+  loginText = 'Log In',
+  signupWithText = 'Or you can Signup with',
+  showSocialIcons = true,
+  showLegalText = true,
+  googleLoginUrl = 'https://accounts.google.com/signin',
+  githubLoginUrl = 'https://github.com/login',
+  facebookLoginUrl = 'https://www.facebook.com/login',
+  privacyPolicyUrl = 'https://policies.google.com/privacy',
+  termsOfServiceUrl = 'https://policies.google.com/terms',
+}) => {
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  const [errors, setErrors] = useState<{ email: string; username: string; password: string }>({
+    email: '',
+    username: '',
+    password: '',
+  });
+
+  const theme = useTheme();
+
+  React.useEffect(() => {
+    if (typeof window === 'undefined') return;
+    const doc = document.documentElement;
+    const body = document.body;
+    const prevDocOverflowX = doc.style.overflowX;
+    const prevBodyOverflowX = body.style.overflowX;
+    doc.style.overflowX = 'hidden';
+    body.style.overflowX = 'hidden';
+    return () => {
+      doc.style.overflowX = prevDocOverflowX || '';
+      body.style.overflowX = prevBodyOverflowX || '';
+    };
+  }, []);
+
+  const handleImgError = async (e: React.SyntheticEvent<HTMLImageElement>) => {
+    const img = e.currentTarget as HTMLImageElement;
+    const srcCandidates: string[] = [];
+    if (img.src) srcCandidates.push(img.src);
+    try {
+      if (img.alt === 'Google') {
+        srcCandidates.push(new URL('./assets/google.svg', import.meta.url).href);
+      }
+      if (img.alt === 'GitHub') {
+        srcCandidates.push(new URL('./assets/github.svg', import.meta.url).href);
+      }
+      if (img.alt === 'Facebook') {
+        srcCandidates.push(new URL('./assets/facebook.svg', import.meta.url).href);
+      }
+    } catch {
+      // ignore
+    }
+
```

Comment:
Additional hardcoded candidate paths (/assets/google.svg, /src/pages/SignUpPage/assets/google.svg) are fragile and may not be available at runtime. Prefer bundler-resolved imports and avoid relying on absolute or repository-relative runtime paths.

Suggestion: fallback to the imported static URLs (GoogleStatic, GithubStatic, FacebookStatic), and only as a last resort consider a known CDN or a documented assets location. Remove or document the use of '/src/...' paths which might not exist in production bundles.

**Sugerencia:**
```typescript
Use imported build-time asset URLs (the '?url' imports) as primary fallback and avoid hard-coded absolute filesystem paths.
```